src/background/index.ts

```diff
@@ -64,6 +64,16 @@ export const headerToToken = async (
             // API expects interactive Challenge at /challenge
             attesterURI = `${attesterURI}/challenge`;
 
+            // To minimize the number of tabs opened, we check if the requesting tab is the active tab
+            const activeTab: number | undefined = await new Promise((resolve) =>
+                chrome.tabs.query({ active: true, currentWindow: true }, (tabs) =>
+                    resolve(tabs[0].id),
+                ),
+            );
+            if (activeTab !== tabId) {
+                logger.debug('Not openning a new tab due to requesting tab not being active');
```

Comment:
typo:
```suggestion
                logger.debug('Not opening a new tab due to requesting tab not being active');
```

Observation: this function returns `undefined`, but the the caller also checks for `null`. Do you purposefully differentiate between them?
```js
        const redirectPromise = w3HeaderValue
            .then(async (value): Promise<void | chrome.webRequest.BlockingResponse> => {
                if (value === null) { // <-- null check
                    delete TOKENS[details.url];
                    return;
                }
                if (isManifestV3(chrome)) {
                   // ...
                } else {
                    if (value) { // <-- undefined check
                        TOKENS[details.url] = value;
                    }
                }

                return { redirectUrl: details.url };
            })
```

Reply:
I return `undefined` because that's the normal return path of similar check in this function. However, this does indeed not seem to be a normal validation.
I've pushed a new commit with the correct check `!value`, which would cover `undefined` and `null` (or even empty string, even though we don't expect this value).

The typo has been corrected as well.

Reply:
With the `!value` change, the second `if (value) {` check is redundant, it will always be true. LGTM otherwise!

Reply:
updated